### PR TITLE
Update opam

### PR DESCRIPTION
**Pull request by yasu100**
- Add without-ocaml option. Without-ocaml option enables to install opam without installation of ocaml and camlp4.

### CODE REVIEW

File: Library/Formula/opam.rb [hunk 1]
```diff
@@ -14,8 +14,12 @@ class Opam < Formula
...
   end
 
-  depends_on "ocaml"
-  depends_on "camlp4" => :recommended
+  option "without-ocaml", "Build without ocaml"
+
+  if build.with? "ocaml"
+    depends_on "ocaml"
+    depends_on "camlp4" => :recommended
+  end
```
Comment (bfontaine) [inline Library/Formula/opam.rb]: Use the following instead:
```rb
depends_on "ocaml" => :recommended
depends_on "camlp4" => :recommended if build.with? "ocaml"
```

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by yasu100** [comment]
Why this check failed?
I tried to read Jenkins log , but there was no result.

**Comment by bfontaine** [comment]
This is a known issue, let’s retry it.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by yasu100** [comment]
What does "CI-requeued" Tag mean?
I saw "All checks have passed".

**Comment by bfontaine** [comment]
@yasu100 It means there was an issue with the tests so we re-queued this PR to be tested. It’s useful so that another maintainer doesn’t try to requeue it another time if they see the tests failed.

**Comment by DomT4** [comment]
@BrewTestBot test this please.

File: Library/Formula/opam.rb [hunk 2]
```diff
@@ -72,23 +72,33 @@ class Opam < Formula
...
+      end
+
+      system "./configure", "--prefix=#{prefix}", "--mandir=#{man}"
+      system "make", "lib-ext"
+      system "make"
+      system "make", "man"
+      system "make", "install"
+
+      bash_completion.install "shell/opam_completion.sh"
+      zsh_completion.install "shell/opam_completion_zsh.sh" => "_opam"
```
Comment (mikemcquaid) [inline Library/Formula/opam.rb]: Please pull the duplicated lines out of the `if`/`else`

**Comment by yasu100** [comment]
Pull the duplicate lines out of the if/else.
Would this be OK?

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `opam: add without-ocaml option.`.

**Pull request closed without merging**